ci/owncloud/upload_to_owncloud.py

```diff
@@ -1,3 +1,47 @@
+from os import path
+from getpass import getuser
 from webdav3.client import Client
 from opts import debug, message, opts
 
+def get_path_components(dirname):
+    """
+    get_path_components(dirname)
+
+    Gets a list of all subpaths needed to build a directory, including itself.
+
+    Parameters
+    ----------
+    dirname : str
+        A given directory path
+    Returns
+    -------
+    A python list of path strings
+    For an input 'a/b/c' returns ['a/b/c', 'a/b', 'a']
+    """
+    path_list = [dirname, ]
+    tmp = path.split(dirname)
+    while tmp[0] != "" and not path.ismount(tmp[0]):
+        path_list.append(tmp[0])
+        tmp = path.split(tmp[0])
+    path_list.reverse()
+    return path_list
+
+
+def mkdir_recursively(webdav_client, remote_path):
+    """
+    mkdir_recursively(remote_path)
+
+    Recursively creates all directories needed to create the input path.
+
+    Parameters
+    ----------
+    remote_path: str
+    """
+    pathlist = get_path_components(remote_path)
+    for p in pathlist:
+        try:
+            webdav_client.mkdir(p)
+        except BaseException as e:
+            debug(e)
```

Comment:
Shouldn't we also raise the exception instead of hiding it?

Reply:
Not exactly, since I'm pretty sure it might throw a 405 status code if the directory already exists, I got a few of those when running tests, so I did this just in case.
Looking at the source code though, It looks like `mkdir` doesn't raise an exception in the case of a 405, I didn't try running the code without the `try` block.

Reply:
Also, you should not catch `BaseException` in python generally speaking. Catch `Exception` instead :-)